flang/lib/Lower/Mangler.cpp

```diff
@@ -126,17 +116,44 @@ Fortran::lower::mangle::mangleName(const Fortran::semantics::Symbol &symbol) {
             return fir::NameUniquer::doCommonBlock(symbolName);
           },
           [&](const Fortran::semantics::DerivedTypeDetails &) {
+            // This is a derived type symbol definition. Note that there are
+            // no kind parameter values associated to it. Kind parameter
+            // values are defined when declaring an entity of this derived type.
```

Comment:
It appears that this lambda is no longer needed and should just be removed. This is mangling a derived type name, derived types can have KIND parameters, and derived types must be mangled with those KIND values. So this is a bug.

Reply:
I was no sure we would never have a use case requiring to mangle the type definition instead of instances, but right now I do not see one, so safer to make it a hard error. Replaced the code by an llvm::report_fatal_error.